bot/bot.py

```diff
@@ -61,8 +76,14 @@ def add_cog(self, cog: commands.Cog) -> None:
 
         This only serves to make the info log, so that extensions don't have to.
         """
-        super().add_cog(cog)
-        log.info(f"Cog loaded: {cog.qualified_name}")
+        cog_name = inspect.getmodule(cog).__name__
+
+        if cog_name in self.unloaded_extensions:
+            self.unloaded_extensions.remove(cog_name)
+            log.info(f"Skipping cog {cog.qualified_name}, found in unloaded cache.")
+        else:
+            super().add_cog(cog)
+            log.info(f"Cog loaded: {cog.qualified_name}")
```

Comment:
Maybe we can just override `load_extension` here

Reply:
Hmm, we could but I don’t see why is that better than this.

Reply:
`Cog.__module__` is not the same as its extension. For example, `InternalEval.__module__` would be `bot.exts.internal_eval._internal_eval`, yet the corresponding extension is `bot.exts.internal_eval`.

Reply:
I am using `Cog.__name__` here not `.__module__`

![Screenshot from 2021-05-22 07-42-11](https://user-images.githubusercontent.com/69356296/119211688-4eecb900-bad1-11eb-95f5-f5a29065752c.png)